Calendar accessibility fixes

lorejoh12 opened the pull request:
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #11508, Fixes #12444, Fixes #13848 
- [x] Include a change request file using `$ yarn change`

#### Description of changes

3 accessibility fixes for Calendar and DatePicker.
- for DatePicker, alt+down arrow should open if the span has focus. Added keydown listener case for it
- for button in header of day picker in Calendar, the button does nothing if the month and day pickers are side by side. We mark the tabIndex = -1 already, but if the user navigates to the header using Narrator's functions, it reports the button as clickable. Changing button to a div so that doesn't happen.
- Added all the various high contrast selectors to match Ben's design for Calendar.

#### Focus areas to test

Can test high contrast only in Win10 + Edge, turn on High Contrast settings from windows settings.

xugao commented on packages/date-time/src/components/CalendarDayGrid/CalendarDayGrid.styles.ts:
```diff
@@ -104,12 +105,33 @@ export const styles = (props: ICalendarDayGridStyleProps): ICalendarDayGridStyle
+      position: 'relative',
       selectors: {
+        [HighContrastSelector]: {
+          color: 'WindowText!important',
+          backgroundColor: 'Window!important',
```
we should avoid setting `!important`. we should use minimal specificity to achieve the goal here. this makes very hard for user to override (not sure if they ever need, but still it's not a good practice in general).

lorejoh12 replied:
I removed a few of them. I run into css specificity issues with merge-styles trying to remove the rest. Specifically, I have a classname `dayCell`, which applies a backgroundColor. I have another class `daySelected`, which applies a different background color. Both classes get put on the same <div>.

The color that actually ends up getting applied depends on the order the classNames end up getting written to the generated CSS. If you open the storybook example, and flip from the Calendar to WeeklyDayPicker examples, you'll see that the order gets flipped for the two classes, whereas I would prefer if `daySelected` styles always win. In pure CSS, you can use a specificity hack like defining `.daySelected.daySelected`, but as far as I know you can't do that in mergestyles. So I end up just putting !important on the backgroundColor, and therefore also put !important on the backgroundColor when defining the high contrast settings.

If you know of a way to override the specificity in merge styles defined in our own component, please let me know and I'll try to remove the other !important tags in a future change.

lorejoh12 commented:
> Actually, all our fixes are made only on `data-time` package. we should make the same fixes in OUFR too? since that's what they are using/bug is filed against.

Fixed. There was some existing high contrast code in OUFR Calendar, I updated it to work in Edge Chromium.

xugao commented on packages/date-time/src/components/CalendarDayGrid/CalendarDayGrid.styles.ts:
```diff
@@ -104,12 +105,46 @@ export const styles = (props: ICalendarDayGridStyleProps): ICalendarDayGridStyle
+        ['&.' + classNames.pressedStyle + '.' + classNames.hoverStyle]: {
+          selectors: {
+            [HighContrastSelector]: {
+              color: 'Highlight',
+              backgroundColor: 'Window',
```
look like you can remove these styles. it's covered by the block above

xugao commented on packages/date-time/src/components/CalendarDayGrid/CalendarDayGrid.styles.ts:
```diff
@@ -104,12 +105,46 @@ export const styles = (props: ICalendarDayGridStyleProps): ICalendarDayGridStyle
+            [HighContrastSelector]: {
+              zIndex: 3,
+              overflow: 'visible',
+              color: 'WindowText',
+              backgroundColor: 'Window',
```
do you need these 2 lines or we can relay on cascaded styles from above

lorejoh12 replied:
can remove color, backgroundColor gets overridden right above this so need to re-override

xugao commented on packages/date-time/src/components/CalendarDayGrid/CalendarDayGrid.styles.ts:
```diff
@@ -104,12 +105,46 @@ export const styles = (props: ICalendarDayGridStyleProps): ICalendarDayGridStyle
+              color: 'WindowText',
+              backgroundColor: 'Window',
+              outline: '1px solid Highlight',
+              MsHighContrastAdjust: 'none',
```
is it possible to move this line from individual blocks to inside
```
 [HighContrastSelector]: {
          color: 'WindowText',
          backgroundColor: 'Window',
          zIndex: 0,
        },
```
